src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,57 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            return null;
+        }
+        if (checkUserAge(user)
+                && checkUserNotRegisteredYet(user)
+                && checkUserLoginLength(user)
+                && checkUserPasswordLength(user)) {
+            storageDao.add(user);
+        }
+        return user;
+    }
+
+    private boolean checkUserNotRegisteredYet(User user) {
+        for (User registeredUser : Storage.people) {
+            if (registeredUser.getLogin() != null && registeredUser.getLogin()
+                    .equals(user.getLogin())) {
+                throw new InvalidUserDataException("This user is already registered!");
+            }
+        }
```

Comment:
Can we use StorageDaoImpl get() method, instead of the for loop? If a user with login already exists in storage, we will get not null User object, other way null, combined with user. login != null, or smth like that?